b2/bucket.py

```diff
@@ -271,6 +275,97 @@ def upload_local_file(
         upload_source = UploadSourceLocalFile(local_path=local_file, content_sha1=sha1_sum)
         return self.upload(upload_source, file_name, content_type, file_infos, progress_listener)
 
+    def upload_stream(
+        self,
+        file_name,
+        part_size,
+        content_type=None,
+        file_infos=None,
+    ):
+        """
+        Uploads a file from stdin stream to a B2 file.
+        """
+        #determine if it requires large file upload, > part_size >= minimum_part_size
+        #TODO make part_size user configurable 100MB to 5GB so max file size of 10TB can be reached with 1 to 10000 parts
+        chunk_size = 16*1024 #16KiB chunk size for memory considerations
+        validate_b2_file_name(file_name)
+        file_info = file_infos or {}
+        content_type = content_type or self.DEFAULT_CONTENT_TYPE
+        progress_listener = DoNothingProgressListener()
+
+        temp_fd, temp_path = tempfile.mkstemp()
+        stdin_reader = io.open(sys.stdin.fileno())
+        stdin_buffered = io.BufferedReader(stdin_reader.buffer) #Allows for peeking stdin
+        with os.fdopen(temp_fd, 'w+b') as tmp:
+            for i in six.moves.xrange(0, part_size, chunk_size):
+                #checks if end of file chunking to be in line with part_size
+                if (i + chunk_size) > part_size:
+                    chunk = stdin_buffered.read(part_size - i)
+                else:
+                    chunk = stdin_buffered.read(chunk_size)
+                if not chunk: #EOF
+                    break
+                tmp.write(chunk)
+            tmp_length = tmp.tell()
+            tmp.seek(0)
+            #check if small file upload
+            #peek to check if stream ended on first part_size boundary
+            if (tmp_length < part_size) or not stdin_buffered.peek(1):
+                #print 'Starting small file upload'
+                upload_source = UploadSourceLocalFile(temp_path)
+                return self._upload_small_file(upload_source, file_name, content_type, file_info, progress_listener)
+            else:
```

Comment:
since the function returns at the end of the `if` block, you could continue the code just after it, which would save the `else:` statement and the indentation that comes after it. That would make the subsequent code easier to read.